// @woocommerce/e2e-utils/build/factories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _api = require("@woocommerce/api");

var _simpleProduct = require("./factories/simple-product");

var _variableProduct = require("./factories/variable-product");

var _variation = require("./factories/variation");

var _groupedProduct = require("./factories/grouped-product");

const config = require('config');

const apiUrl = config.get('url');
const adminUsername = config.get('users.admin.username');
const adminPassword = config.get('users.admin.password');

const withDefaultPermalinks = _api.HTTPClientFactory.build(apiUrl).withBasicAuth(adminUsername, adminPassword).create();

const withIndexPermalinks = _api.HTTPClientFactory.build(apiUrl).withBasicAuth(adminUsername, adminPassword).withIndexPermalinks().create();

const factories = {
  api: {
    withDefaultPermalinks,
    withIndexPermalinks
  },
  products: {
    simple: (0, _simpleProduct.simpleProductFactory)(withDefaultPermalinks),
    variable: (0, _variableProduct.variableProductFactory)(withDefaultPermalinks),
    variation: (0, _variation.variationFactory)(withDefaultPermalinks),
    grouped: (0, _groupedProduct.groupedProductFactory)(withDefaultPermalinks)
  }
};
var _default = factories;
exports.default = _default;
