// @woocommerce/e2e-utils/build/flows/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WP_ADMIN_WP_UPDATES = exports.WP_ADMIN_WC_SETTINGS = exports.WP_ADMIN_WC_HOME = exports.WP_ADMIN_WC_HELPER = exports.WP_ADMIN_WC_EXTENSIONS = exports.WP_ADMIN_SETUP_WIZARD = exports.WP_ADMIN_POST_TYPE = exports.WP_ADMIN_PLUGIN_PAGE = exports.WP_ADMIN_PLUGIN_INSTALL = exports.WP_ADMIN_PLUGINS = exports.WP_ADMIN_PERMALINK_SETTINGS = exports.WP_ADMIN_NEW_SHIPPING_ZONE = exports.WP_ADMIN_NEW_PRODUCT = exports.WP_ADMIN_NEW_POST_TYPE = exports.WP_ADMIN_NEW_ORDER = exports.WP_ADMIN_NEW_COUPON = exports.WP_ADMIN_LOGIN = exports.WP_ADMIN_IMPORT_PRODUCTS = exports.WP_ADMIN_DASHBOARD = exports.WP_ADMIN_ANALYTICS_PAGES = exports.WP_ADMIN_ALL_USERS_VIEW = exports.WP_ADMIN_ALL_PRODUCTS_VIEW = exports.WP_ADMIN_ALL_ORDERS_VIEW = exports.WP_ADMIN_ALL_COUPONS_VIEW = exports.SHOP_PRODUCT_PAGE = exports.SHOP_PAGE = exports.SHOP_MY_ACCOUNT_PAGE = exports.SHOP_CHECKOUT_PAGE = exports.SHOP_CART_PAGE = exports.MY_ACCOUNT_ORDERS = exports.MY_ACCOUNT_DOWNLOADS = exports.MY_ACCOUNT_ADDRESSES = exports.MY_ACCOUNT_ACCOUNT_DETAILS = exports.IS_RETEST_MODE = void 0;

/**
 * External dependencies
 */
const config = require('config');

const baseUrl = config.get('url');
/**
 * WordPress core dashboard pages.
 * @type {string}
 */

const WP_ADMIN_LOGIN = baseUrl + 'wp-login.php';
exports.WP_ADMIN_LOGIN = WP_ADMIN_LOGIN;
const WP_ADMIN_DASHBOARD = baseUrl + 'wp-admin/';
exports.WP_ADMIN_DASHBOARD = WP_ADMIN_DASHBOARD;
const WP_ADMIN_WP_UPDATES = WP_ADMIN_DASHBOARD + 'update-core.php';
exports.WP_ADMIN_WP_UPDATES = WP_ADMIN_WP_UPDATES;
const WP_ADMIN_PLUGINS = WP_ADMIN_DASHBOARD + 'plugins.php';
exports.WP_ADMIN_PLUGINS = WP_ADMIN_PLUGINS;
const WP_ADMIN_PLUGIN_INSTALL = WP_ADMIN_DASHBOARD + 'plugin-install.php';
exports.WP_ADMIN_PLUGIN_INSTALL = WP_ADMIN_PLUGIN_INSTALL;
const WP_ADMIN_PERMALINK_SETTINGS = WP_ADMIN_DASHBOARD + 'options-permalink.php';
exports.WP_ADMIN_PERMALINK_SETTINGS = WP_ADMIN_PERMALINK_SETTINGS;
const WP_ADMIN_ALL_USERS_VIEW = WP_ADMIN_DASHBOARD + 'users.php';
/**
 * WooCommerce core post type pages.
 * @type {string}
 */

exports.WP_ADMIN_ALL_USERS_VIEW = WP_ADMIN_ALL_USERS_VIEW;
const WP_ADMIN_POST_TYPE = WP_ADMIN_DASHBOARD + 'edit.php?post_type=';
exports.WP_ADMIN_POST_TYPE = WP_ADMIN_POST_TYPE;
const WP_ADMIN_NEW_POST_TYPE = WP_ADMIN_DASHBOARD + 'post-new.php?post_type=';
exports.WP_ADMIN_NEW_POST_TYPE = WP_ADMIN_NEW_POST_TYPE;
const WP_ADMIN_ALL_COUPONS_VIEW = WP_ADMIN_POST_TYPE + 'shop_coupon';
exports.WP_ADMIN_ALL_COUPONS_VIEW = WP_ADMIN_ALL_COUPONS_VIEW;
const WP_ADMIN_NEW_COUPON = WP_ADMIN_NEW_POST_TYPE + 'shop_coupon';
exports.WP_ADMIN_NEW_COUPON = WP_ADMIN_NEW_COUPON;
const WP_ADMIN_ALL_ORDERS_VIEW = WP_ADMIN_POST_TYPE + 'shop_order';
exports.WP_ADMIN_ALL_ORDERS_VIEW = WP_ADMIN_ALL_ORDERS_VIEW;
const WP_ADMIN_NEW_ORDER = WP_ADMIN_NEW_POST_TYPE + 'shop_order';
exports.WP_ADMIN_NEW_ORDER = WP_ADMIN_NEW_ORDER;
const WP_ADMIN_ALL_PRODUCTS_VIEW = WP_ADMIN_POST_TYPE + 'product';
exports.WP_ADMIN_ALL_PRODUCTS_VIEW = WP_ADMIN_ALL_PRODUCTS_VIEW;
const WP_ADMIN_NEW_PRODUCT = WP_ADMIN_NEW_POST_TYPE + 'product';
exports.WP_ADMIN_NEW_PRODUCT = WP_ADMIN_NEW_PRODUCT;
const WP_ADMIN_IMPORT_PRODUCTS = WP_ADMIN_ALL_PRODUCTS_VIEW + '&page=product_importer';
/**
 * WooCommerce settings pages.
 * @type {string}
 */

exports.WP_ADMIN_IMPORT_PRODUCTS = WP_ADMIN_IMPORT_PRODUCTS;
const WP_ADMIN_PLUGIN_PAGE = WP_ADMIN_DASHBOARD + 'admin.php?page=';
exports.WP_ADMIN_PLUGIN_PAGE = WP_ADMIN_PLUGIN_PAGE;
const WP_ADMIN_WC_HOME = WP_ADMIN_PLUGIN_PAGE + 'wc-admin';
exports.WP_ADMIN_WC_HOME = WP_ADMIN_WC_HOME;
const WP_ADMIN_SETUP_WIZARD = WP_ADMIN_WC_HOME + '&path=%2Fsetup-wizard';
exports.WP_ADMIN_SETUP_WIZARD = WP_ADMIN_SETUP_WIZARD;
const WP_ADMIN_ANALYTICS_PAGES = WP_ADMIN_WC_HOME + '&path=%2Fanalytics%2F';
exports.WP_ADMIN_ANALYTICS_PAGES = WP_ADMIN_ANALYTICS_PAGES;
const WP_ADMIN_WC_SETTINGS = WP_ADMIN_PLUGIN_PAGE + 'wc-settings&tab=';
exports.WP_ADMIN_WC_SETTINGS = WP_ADMIN_WC_SETTINGS;
const WP_ADMIN_NEW_SHIPPING_ZONE = WP_ADMIN_WC_SETTINGS + 'shipping&zone_id=new';
exports.WP_ADMIN_NEW_SHIPPING_ZONE = WP_ADMIN_NEW_SHIPPING_ZONE;
const WP_ADMIN_WC_EXTENSIONS = WP_ADMIN_PLUGIN_PAGE + 'wc-addons';
exports.WP_ADMIN_WC_EXTENSIONS = WP_ADMIN_WC_EXTENSIONS;
const WP_ADMIN_WC_HELPER = WP_ADMIN_PLUGIN_PAGE + 'wc-addons&section=helper';
/**
 * Shop pages.
 * @type {string}
 */

exports.WP_ADMIN_WC_HELPER = WP_ADMIN_WC_HELPER;
const SHOP_PAGE = baseUrl + 'shop';
exports.SHOP_PAGE = SHOP_PAGE;
const SHOP_PRODUCT_PAGE = baseUrl + '?p=';
exports.SHOP_PRODUCT_PAGE = SHOP_PRODUCT_PAGE;
const SHOP_CART_PAGE = baseUrl + 'cart';
exports.SHOP_CART_PAGE = SHOP_CART_PAGE;
const SHOP_CHECKOUT_PAGE = baseUrl + 'checkout/';
exports.SHOP_CHECKOUT_PAGE = SHOP_CHECKOUT_PAGE;
const SHOP_MY_ACCOUNT_PAGE = baseUrl + 'my-account/';
/**
 * Customer account pages.
 * @type {string}
 */

exports.SHOP_MY_ACCOUNT_PAGE = SHOP_MY_ACCOUNT_PAGE;
const MY_ACCOUNT_ORDERS = SHOP_MY_ACCOUNT_PAGE + 'orders';
exports.MY_ACCOUNT_ORDERS = MY_ACCOUNT_ORDERS;
const MY_ACCOUNT_DOWNLOADS = SHOP_MY_ACCOUNT_PAGE + 'downloads';
exports.MY_ACCOUNT_DOWNLOADS = MY_ACCOUNT_DOWNLOADS;
const MY_ACCOUNT_ADDRESSES = SHOP_MY_ACCOUNT_PAGE + 'edit-address';
exports.MY_ACCOUNT_ADDRESSES = MY_ACCOUNT_ADDRESSES;
const MY_ACCOUNT_ACCOUNT_DETAILS = SHOP_MY_ACCOUNT_PAGE + 'edit-account';
/**
 * Test control flags.
 * @type {boolean}
 */

exports.MY_ACCOUNT_ACCOUNT_DETAILS = MY_ACCOUNT_ACCOUNT_DETAILS;
const IS_RETEST_MODE = process.env.E2E_RETEST == '1';
exports.IS_RETEST_MODE = IS_RETEST_MODE;
