// @wordpress/e2e-test-utils/build/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  activatePlugin: true,
  activateTheme: true,
  arePrePublishChecksEnabled: true,
  changeSiteTimezone: true,
  clearLocalStorage: true,
  clickBlockAppender: true,
  clickBlockToolbarButton: true,
  clickButton: true,
  clickMenuItem: true,
  clickOnCloseModalButton: true,
  clickOnMoreMenuItem: true,
  createNewPost: true,
  createURL: true,
  deactivatePlugin: true,
  deleteTheme: true,
  disablePrePublishChecks: true,
  dragAndResize: true,
  enablePageDialogAccept: true,
  enablePrePublishChecks: true,
  ensureSidebarOpened: true,
  findSidebarPanelToggleButtonWithTitle: true,
  findSidebarPanelWithTitle: true,
  getAllBlockInserterItemTitles: true,
  getAllBlocks: true,
  getAvailableBlockTransforms: true,
  getBlockSetting: true,
  getEditedPostContent: true,
  getCurrentPostContent: true,
  hasBlockSwitcher: true,
  getPageError: true,
  insertBlock: true,
  insertPattern: true,
  insertReusableBlock: true,
  searchForBlock: true,
  searchForPattern: true,
  searchForReusableBlock: true,
  insertBlockDirectoryBlock: true,
  openGlobalBlockInserter: true,
  closeGlobalBlockInserter: true,
  toggleGlobalBlockInserter: true,
  installPlugin: true,
  installTheme: true,
  isCurrentURL: true,
  isInDefaultBlock: true,
  loginUser: true,
  enableFocusLossObservation: true,
  disableFocusLossObservation: true,
  openDocumentSettingsSidebar: true,
  openPublishPanel: true,
  trashAllPosts: true,
  pressKeyTimes: true,
  pressKeyWithModifier: true,
  setClipboardData: true,
  publishPost: true,
  publishPostWithPrePublishChecksDisabled: true,
  saveDraft: true,
  selectBlockByClientId: true,
  setBrowserViewport: true,
  setPostContent: true,
  switchEditorModeTo: true,
  switchUserToAdmin: true,
  switchUserToTest: true,
  isThemeInstalled: true,
  toggleMoreMenu: true,
  toggleOfflineMode: true,
  isOfflineMode: true,
  toggleScreenOption: true,
  transformBlockTo: true,
  uninstallPlugin: true,
  visitAdminPage: true,
  waitForWindowDimensions: true,
  showBlockToolbar: true,
  openPreviewPage: true
};
Object.defineProperty(exports, "activatePlugin", {
  enumerable: true,
  get: function get() {
    return _activatePlugin.activatePlugin;
  }
});
Object.defineProperty(exports, "activateTheme", {
  enumerable: true,
  get: function get() {
    return _activateTheme.activateTheme;
  }
});
Object.defineProperty(exports, "arePrePublishChecksEnabled", {
  enumerable: true,
  get: function get() {
    return _arePrePublishChecksEnabled.arePrePublishChecksEnabled;
  }
});
Object.defineProperty(exports, "changeSiteTimezone", {
  enumerable: true,
  get: function get() {
    return _changeSiteTimezone.changeSiteTimezone;
  }
});
Object.defineProperty(exports, "clearLocalStorage", {
  enumerable: true,
  get: function get() {
    return _clearLocalStorage.clearLocalStorage;
  }
});
Object.defineProperty(exports, "clickBlockAppender", {
  enumerable: true,
  get: function get() {
    return _clickBlockAppender.clickBlockAppender;
  }
});
Object.defineProperty(exports, "clickBlockToolbarButton", {
  enumerable: true,
  get: function get() {
    return _clickBlockToolbarButton.clickBlockToolbarButton;
  }
});
Object.defineProperty(exports, "clickButton", {
  enumerable: true,
  get: function get() {
    return _clickButton.clickButton;
  }
});
Object.defineProperty(exports, "clickMenuItem", {
  enumerable: true,
  get: function get() {
    return _clickMenuItem.clickMenuItem;
  }
});
Object.defineProperty(exports, "clickOnCloseModalButton", {
  enumerable: true,
  get: function get() {
    return _clickOnCloseModalButton.clickOnCloseModalButton;
  }
});
Object.defineProperty(exports, "clickOnMoreMenuItem", {
  enumerable: true,
  get: function get() {
    return _clickOnMoreMenuItem.clickOnMoreMenuItem;
  }
});
Object.defineProperty(exports, "createNewPost", {
  enumerable: true,
  get: function get() {
    return _createNewPost.createNewPost;
  }
});
Object.defineProperty(exports, "createURL", {
  enumerable: true,
  get: function get() {
    return _createUrl.createURL;
  }
});
Object.defineProperty(exports, "deactivatePlugin", {
  enumerable: true,
  get: function get() {
    return _deactivatePlugin.deactivatePlugin;
  }
});
Object.defineProperty(exports, "deleteTheme", {
  enumerable: true,
  get: function get() {
    return _deleteTheme.deleteTheme;
  }
});
Object.defineProperty(exports, "disablePrePublishChecks", {
  enumerable: true,
  get: function get() {
    return _disablePrePublishChecks.disablePrePublishChecks;
  }
});
Object.defineProperty(exports, "dragAndResize", {
  enumerable: true,
  get: function get() {
    return _dragAndResize.dragAndResize;
  }
});
Object.defineProperty(exports, "enablePageDialogAccept", {
  enumerable: true,
  get: function get() {
    return _enablePageDialogAccept.enablePageDialogAccept;
  }
});
Object.defineProperty(exports, "enablePrePublishChecks", {
  enumerable: true,
  get: function get() {
    return _enablePrePublishChecks.enablePrePublishChecks;
  }
});
Object.defineProperty(exports, "ensureSidebarOpened", {
  enumerable: true,
  get: function get() {
    return _ensureSidebarOpened.ensureSidebarOpened;
  }
});
Object.defineProperty(exports, "findSidebarPanelToggleButtonWithTitle", {
  enumerable: true,
  get: function get() {
    return _findSidebarPanelToggleButtonWithTitle.findSidebarPanelToggleButtonWithTitle;
  }
});
Object.defineProperty(exports, "findSidebarPanelWithTitle", {
  enumerable: true,
  get: function get() {
    return _findSidebarPanelWithTitle.findSidebarPanelWithTitle;
  }
});
Object.defineProperty(exports, "getAllBlockInserterItemTitles", {
  enumerable: true,
  get: function get() {
    return _getAllBlockInserterItemTitles.getAllBlockInserterItemTitles;
  }
});
Object.defineProperty(exports, "getAllBlocks", {
  enumerable: true,
  get: function get() {
    return _getAllBlocks.getAllBlocks;
  }
});
Object.defineProperty(exports, "getAvailableBlockTransforms", {
  enumerable: true,
  get: function get() {
    return _getAvailableBlockTransforms.getAvailableBlockTransforms;
  }
});
Object.defineProperty(exports, "getBlockSetting", {
  enumerable: true,
  get: function get() {
    return _getBlockSetting.getBlockSetting;
  }
});
Object.defineProperty(exports, "getEditedPostContent", {
  enumerable: true,
  get: function get() {
    return _getEditedPostContent.getEditedPostContent;
  }
});
Object.defineProperty(exports, "getCurrentPostContent", {
  enumerable: true,
  get: function get() {
    return _getCurrentPostContent.getCurrentPostContent;
  }
});
Object.defineProperty(exports, "hasBlockSwitcher", {
  enumerable: true,
  get: function get() {
    return _hasBlockSwitcher.hasBlockSwitcher;
  }
});
Object.defineProperty(exports, "getPageError", {
  enumerable: true,
  get: function get() {
    return _getPageError.getPageError;
  }
});
Object.defineProperty(exports, "insertBlock", {
  enumerable: true,
  get: function get() {
    return _inserter.insertBlock;
  }
});
Object.defineProperty(exports, "insertPattern", {
  enumerable: true,
  get: function get() {
    return _inserter.insertPattern;
  }
});
Object.defineProperty(exports, "insertReusableBlock", {
  enumerable: true,
  get: function get() {
    return _inserter.insertReusableBlock;
  }
});
Object.defineProperty(exports, "searchForBlock", {
  enumerable: true,
  get: function get() {
    return _inserter.searchForBlock;
  }
});
Object.defineProperty(exports, "searchForPattern", {
  enumerable: true,
  get: function get() {
    return _inserter.searchForPattern;
  }
});
Object.defineProperty(exports, "searchForReusableBlock", {
  enumerable: true,
  get: function get() {
    return _inserter.searchForReusableBlock;
  }
});
Object.defineProperty(exports, "insertBlockDirectoryBlock", {
  enumerable: true,
  get: function get() {
    return _inserter.insertBlockDirectoryBlock;
  }
});
Object.defineProperty(exports, "openGlobalBlockInserter", {
  enumerable: true,
  get: function get() {
    return _inserter.openGlobalBlockInserter;
  }
});
Object.defineProperty(exports, "closeGlobalBlockInserter", {
  enumerable: true,
  get: function get() {
    return _inserter.closeGlobalBlockInserter;
  }
});
Object.defineProperty(exports, "toggleGlobalBlockInserter", {
  enumerable: true,
  get: function get() {
    return _inserter.toggleGlobalBlockInserter;
  }
});
Object.defineProperty(exports, "installPlugin", {
  enumerable: true,
  get: function get() {
    return _installPlugin.installPlugin;
  }
});
Object.defineProperty(exports, "installTheme", {
  enumerable: true,
  get: function get() {
    return _installTheme.installTheme;
  }
});
Object.defineProperty(exports, "isCurrentURL", {
  enumerable: true,
  get: function get() {
    return _isCurrentUrl.isCurrentURL;
  }
});
Object.defineProperty(exports, "isInDefaultBlock", {
  enumerable: true,
  get: function get() {
    return _isInDefaultBlock.isInDefaultBlock;
  }
});
Object.defineProperty(exports, "loginUser", {
  enumerable: true,
  get: function get() {
    return _loginUser.loginUser;
  }
});
Object.defineProperty(exports, "enableFocusLossObservation", {
  enumerable: true,
  get: function get() {
    return _observeFocusLoss.enableFocusLossObservation;
  }
});
Object.defineProperty(exports, "disableFocusLossObservation", {
  enumerable: true,
  get: function get() {
    return _observeFocusLoss.disableFocusLossObservation;
  }
});
Object.defineProperty(exports, "openDocumentSettingsSidebar", {
  enumerable: true,
  get: function get() {
    return _openDocumentSettingsSidebar.openDocumentSettingsSidebar;
  }
});
Object.defineProperty(exports, "openPublishPanel", {
  enumerable: true,
  get: function get() {
    return _openPublishPanel.openPublishPanel;
  }
});
Object.defineProperty(exports, "trashAllPosts", {
  enumerable: true,
  get: function get() {
    return _posts.trashAllPosts;
  }
});
Object.defineProperty(exports, "pressKeyTimes", {
  enumerable: true,
  get: function get() {
    return _pressKeyTimes.pressKeyTimes;
  }
});
Object.defineProperty(exports, "pressKeyWithModifier", {
  enumerable: true,
  get: function get() {
    return _pressKeyWithModifier.pressKeyWithModifier;
  }
});
Object.defineProperty(exports, "setClipboardData", {
  enumerable: true,
  get: function get() {
    return _pressKeyWithModifier.setClipboardData;
  }
});
Object.defineProperty(exports, "publishPost", {
  enumerable: true,
  get: function get() {
    return _publishPost.publishPost;
  }
});
Object.defineProperty(exports, "publishPostWithPrePublishChecksDisabled", {
  enumerable: true,
  get: function get() {
    return _publishPostWithPrePublishChecksDisabled.publishPostWithPrePublishChecksDisabled;
  }
});
Object.defineProperty(exports, "saveDraft", {
  enumerable: true,
  get: function get() {
    return _saveDraft.saveDraft;
  }
});
Object.defineProperty(exports, "selectBlockByClientId", {
  enumerable: true,
  get: function get() {
    return _selectBlockByClientId.selectBlockByClientId;
  }
});
Object.defineProperty(exports, "setBrowserViewport", {
  enumerable: true,
  get: function get() {
    return _setBrowserViewport.setBrowserViewport;
  }
});
Object.defineProperty(exports, "setPostContent", {
  enumerable: true,
  get: function get() {
    return _setPostContent.setPostContent;
  }
});
Object.defineProperty(exports, "switchEditorModeTo", {
  enumerable: true,
  get: function get() {
    return _switchEditorModeTo.switchEditorModeTo;
  }
});
Object.defineProperty(exports, "switchUserToAdmin", {
  enumerable: true,
  get: function get() {
    return _switchUserToAdmin.switchUserToAdmin;
  }
});
Object.defineProperty(exports, "switchUserToTest", {
  enumerable: true,
  get: function get() {
    return _switchUserToTest.switchUserToTest;
  }
});
Object.defineProperty(exports, "isThemeInstalled", {
  enumerable: true,
  get: function get() {
    return _themeInstalled.isThemeInstalled;
  }
});
Object.defineProperty(exports, "toggleMoreMenu", {
  enumerable: true,
  get: function get() {
    return _toggleMoreMenu.toggleMoreMenu;
  }
});
Object.defineProperty(exports, "toggleOfflineMode", {
  enumerable: true,
  get: function get() {
    return _offlineMode.toggleOfflineMode;
  }
});
Object.defineProperty(exports, "isOfflineMode", {
  enumerable: true,
  get: function get() {
    return _offlineMode.isOfflineMode;
  }
});
Object.defineProperty(exports, "toggleScreenOption", {
  enumerable: true,
  get: function get() {
    return _toggleScreenOption.toggleScreenOption;
  }
});
Object.defineProperty(exports, "transformBlockTo", {
  enumerable: true,
  get: function get() {
    return _transformBlockTo.transformBlockTo;
  }
});
Object.defineProperty(exports, "uninstallPlugin", {
  enumerable: true,
  get: function get() {
    return _uninstallPlugin.uninstallPlugin;
  }
});
Object.defineProperty(exports, "visitAdminPage", {
  enumerable: true,
  get: function get() {
    return _visitAdminPage.visitAdminPage;
  }
});
Object.defineProperty(exports, "waitForWindowDimensions", {
  enumerable: true,
  get: function get() {
    return _waitForWindowDimensions.waitForWindowDimensions;
  }
});
Object.defineProperty(exports, "showBlockToolbar", {
  enumerable: true,
  get: function get() {
    return _showBlockToolbar.showBlockToolbar;
  }
});
Object.defineProperty(exports, "openPreviewPage", {
  enumerable: true,
  get: function get() {
    return _preview.openPreviewPage;
  }
});

var _activatePlugin = require("./activate-plugin");

var _activateTheme = require("./activate-theme");

var _arePrePublishChecksEnabled = require("./are-pre-publish-checks-enabled");

var _changeSiteTimezone = require("./change-site-timezone");

var _clearLocalStorage = require("./clear-local-storage");

var _clickBlockAppender = require("./click-block-appender");

var _clickBlockToolbarButton = require("./click-block-toolbar-button");

var _clickButton = require("./click-button");

var _clickMenuItem = require("./click-menu-item");

var _clickOnCloseModalButton = require("./click-on-close-modal-button");

var _clickOnMoreMenuItem = require("./click-on-more-menu-item");

var _createNewPost = require("./create-new-post");

var _createUrl = require("./create-url");

var _deactivatePlugin = require("./deactivate-plugin");

var _deleteTheme = require("./delete-theme");

var _disablePrePublishChecks = require("./disable-pre-publish-checks");

var _dragAndResize = require("./drag-and-resize");

var _enablePageDialogAccept = require("./enable-page-dialog-accept");

var _enablePrePublishChecks = require("./enable-pre-publish-checks");

var _ensureSidebarOpened = require("./ensure-sidebar-opened");

var _findSidebarPanelToggleButtonWithTitle = require("./find-sidebar-panel-toggle-button-with-title");

var _findSidebarPanelWithTitle = require("./find-sidebar-panel-with-title");

var _getAllBlockInserterItemTitles = require("./get-all-block-inserter-item-titles");

var _getAllBlocks = require("./get-all-blocks");

var _getAvailableBlockTransforms = require("./get-available-block-transforms");

var _getBlockSetting = require("./get-block-setting");

var _getEditedPostContent = require("./get-edited-post-content");

var _getCurrentPostContent = require("./get-current-post-content");

var _hasBlockSwitcher = require("./has-block-switcher");

var _getPageError = require("./get-page-error");

var _inserter = require("./inserter");

var _installPlugin = require("./install-plugin");

var _installTheme = require("./install-theme");

var _isCurrentUrl = require("./is-current-url");

var _isInDefaultBlock = require("./is-in-default-block");

var _loginUser = require("./login-user");

var _observeFocusLoss = require("./observe-focus-loss");

var _openDocumentSettingsSidebar = require("./open-document-settings-sidebar");

var _openPublishPanel = require("./open-publish-panel");

var _posts = require("./posts");

var _pressKeyTimes = require("./press-key-times");

var _pressKeyWithModifier = require("./press-key-with-modifier");

var _publishPost = require("./publish-post");

var _publishPostWithPrePublishChecksDisabled = require("./publish-post-with-pre-publish-checks-disabled");

var _saveDraft = require("./save-draft");

var _selectBlockByClientId = require("./select-block-by-client-id");

var _setBrowserViewport = require("./set-browser-viewport");

var _setPostContent = require("./set-post-content");

var _switchEditorModeTo = require("./switch-editor-mode-to");

var _switchUserToAdmin = require("./switch-user-to-admin");

var _switchUserToTest = require("./switch-user-to-test");

var _themeInstalled = require("./theme-installed");

var _toggleMoreMenu = require("./toggle-more-menu");

var _offlineMode = require("./offline-mode");

var _toggleScreenOption = require("./toggle-screen-option");

var _transformBlockTo = require("./transform-block-to");

var _uninstallPlugin = require("./uninstall-plugin");

var _visitAdminPage = require("./visit-admin-page");

var _waitForWindowDimensions = require("./wait-for-window-dimensions");

var _showBlockToolbar = require("./show-block-toolbar");

var _preview = require("./preview");

var _mocks = require("./mocks");

Object.keys(_mocks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _mocks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _mocks[key];
    }
  });
});
